tests/kafkatest/tests/streams/streams_eos_test.py

```diff
@@ -58,16 +58,16 @@ def run_rebalance(self, processor1, processor2, processor3, verifier):
 
         self.driver.start()
 
-        processor1.clean_node_enabled = False
-
         self.add_streams(processor1)
+        processor1.clean_node_enabled = False
         self.add_streams2(processor1, processor2)
         self.add_streams3(processor1, processor2, processor3)
         self.stop_streams3(processor2, processor3, processor1)
         self.add_streams3(processor2, processor3, processor1)
         self.stop_streams3(processor1, processor3, processor2)
         self.stop_streams2(processor1, processor3)
         self.stop_streams(processor1)
+        processor1.clean_node_enabled = True
```

Comment:
We can set the clean-up flag here again to ensure that the node is cleaned when the test is torn down.